minio/thread_pool.py

```diff
@@ -42,11 +42,17 @@ def __init__(self, tasks_queue, results_queue, exceptions_queue):
         self.start()
 
     def run(self):
+        """ Continously receive tasks and execute them """
         fast_quit = False
-        while not self.tasks_queue.empty():
-            func, args, kargs = self.tasks_queue.get()
+        while True:
```

Comment:
When a task throws an exception, `fast_quit` is set to `True`. IIUC, the loop proceeds to the next iteration, 'pops' the next task out of the queue __but__ exits the loop without executing it. So, the 'popped' task will __not__ be run by any of the workers in the thread pool? Is this behaviour acceptable?

Reply:
> When a task throws an exception, `fast_quit` is set to `True`. IIUC, the loop proceeds to the next iteration, 'pops' the next task out of the queue **but** exits the loop without executing it. So, the 'popped' task will **not** be run by any of the workers in the thread pool? Is this behaviour acceptable?

Yes, if one thread has an exception, it is not needed for other threads to continue uploads of parts

Reply:
> Yes, if one thread has an exception, it is not needed for other threads to continue uploads of parts

I will modify the code a little bit, only the current thread is quitting, not other threads as I thought

Reply:
> I will modify the code a little bit, only the current thread is quitting, not other threads as I thought

I removed fast_quit variable, now we will test directly on exceptions_queue and see if it is empty before continuing to run tasks